src/contracts/entrypoints.rs

```diff
@@ -91,22 +199,69 @@ pub trait Invariant {
         pool_key: PoolKey,
     ) -> Result<(), InvariantError>;
 
+    /// Allows an authorized user to claim collected fees.
+    ///
+    /// # Parameters
+    /// - `index`: The index of the user position from which fees will be claimed.
+    ///
+    /// # Errors
+    /// - Fails if the position cannot be found.
+    /// - Fails if the DEX has insufficient balance to perform the transfer.
     #[ink(message)]
     fn claim_fee(&mut self, index: u32) -> Result<(TokenAmount, TokenAmount), InvariantError>;
 
+    /// Removes a position.
+    ///
+    /// # Parameters
+    /// - `index`: The index of the user position to be removed.
+    ///
+    /// # Events
+    /// - Emits a `Remove Position` event upon success.
+    ///
+    /// # Errors
+    /// - Fails if unable to deinitialize ticks.
+    /// - Fails if the DEX has insufficient balance to perform the transfer.
     #[ink(message)]
     fn remove_position(&mut self, index: u32)
         -> Result<(TokenAmount, TokenAmount), InvariantError>;
 
+    /// Allows a user to add a custom fee tier.
+    ///
+    /// # Parameters
+    /// - `fee_tier`: A struct identifying the pool fee and tick spacing.
+    ///
+    /// # Errors
+    /// - Fails if an unauthorized user attempts to create a fee tier.
+    /// - Fails if the tick spacing is invalid.
+    /// - Fails if the fee tier already exists.
     #[ink(message)]
     fn add_fee_tier(&mut self, fee_tier: FeeTier) -> Result<(), InvariantError>;
 
+    /// Retrieves information about a fee tier.
+    ///
+    /// # Parameters
+    /// - `fee_tier_key`: A struct identifying the pool fee and tick spacing.
     #[ink(message)]
     fn get_fee_tier(&self, key: FeeTierKey) -> Option<()>;
 
+    /// Removes an existing fee tier.
+    ///
+    /// # Parameters
+    /// - `fee_tier_key`: A struct identifying the pool fee and tick spacing.
     #[ink(message)]
     fn remove_fee_tier(&mut self, key: FeeTierKey);
 
+    /// Allows a user to create a custom pool on a specified token pair and fee tier.
+    ///
+    /// # Parameters
+    /// - `token_0`: The address of the first token.
+    /// - `token_1`: The address of the second token.
+    /// - `fee_tier`: A struct identifying the pool fee and tick spacing.
+    /// - `init_tick`: The initial tick at which the pool will be created.
+    ///
+    /// # Errors
+    /// - Fails if the specified fee tier cannot be found.
+    /// - Fails if the user attempts to create a pool for the same tokens.
```

Comment:
should fail if pool with the same tokens and fee tier already exist (you should also refactor create pool message)